Description: Fix a postconf segfault when map file cannot be read
Author: Viktor Dukhovni <postfix-users@dukhovni.org>
Applied-Upstream: https://archive.mgm51.com/mirrors/postfix-source/official/postfix-3.3.1.tar.gz
Origin: https://marc.info/?l=postfix-users&m=152578771531514&w=2
Bug-Debian: https://bugs.debian.org/898271
Bug-Ubuntu: https://launchpad.net/bugs/1753470
Last-Update: 2018-08-20
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/postconf/postconf_dbms.c
+++ b/src/postconf/postconf_dbms.c
@@ -174,10 +174,10 @@
 	 */
 	dict = dict_ht_open(dict_spec, O_CREAT | O_RDWR, 0);
 	dict_register(dict_spec, dict);
-	if ((fp = vstream_fopen(cf_file, O_RDONLY, 0)) == 0
-	    && errno != EACCES) {
-	    msg_warn("open \"%s\" configuration \"%s\": %m",
-		     dp->db_type, cf_file);
+	if ((fp = vstream_fopen(cf_file, O_RDONLY, 0)) == 0) {
+        if (errno != EACCES)
+	        msg_warn("open \"%s\" configuration \"%s\": %m",
+		             dp->db_type, cf_file);
 	    myfree(dict_spec);
 	    return;
 	}
